lib/gooddata/models/metadata/analytical_visualization_object.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: UTF-8
+# frozen_string_literal: true
+#
+# Copyright (c) 2010-2021 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+module GoodData
+  class AnalyticalVisualizationObject < GoodData::MdObject
+    class << self
+      # Create a specify object in the specify project
+      #
+      # @param object_data [Hash] the data of object will be created
+      # @param klass [Class] A class used for instantiating the returned data
+      # @param empty_data_object [Hash] the empty data of object will be created
+      # @param assignable_properties [Hash] the properties allow updating
+      # @param options [Hash] The project that the object will be created in
+      # @return klass object
+      def create(object_data, klass, empty_data_object = {}, assignable_properties = [], options = { :client => GoodData.client, :project => GoodData.project })
+        client, project = GoodData.get_client_and_project(GoodData::Helpers.symbolize_keys(options))
+
+        res = client.create(klass, GoodData::Helpers.deep_dup(GoodData::Helpers.stringify_keys(empty_data_object)), :project => project)
+        object_data.each do |k, v|
+          res.send("#{k}=", v) if assignable_properties.include? k
+        end
+        res
+      end
+    end
+
+    # Method used for replacing values in their state according to mapping.
+    # Can be used to replace any values but it is typically used to replace
+    # the URIs. Returns a new object of the same type.
+    #
+    # @param [Array<Array>]mapping specifying what should be exchanged for what. As mapping should be used output of GoodData::Helpers.prepare_mapping.
+    # @return [GoodData::AnalyticalVisualizationObject]
+    def replace(mapping)
```

Comment:
We should have unit test for this function.

Reply:
Checked the function **replace(mapping)** https://github.com/gooddata/gooddata-ruby/blob/develop/lib/gooddata/models/metadata.rb#L199 in parent class already cover same as this function. So delete this method and no need any unit tests for this function.